### PR TITLE
#198 NPE in withBody()

### PR DESCRIPTION
**Pull request by alevohin**
Added stacktrace loggng

### CODE REVIEW

**Comment by davvd** [comment]
Many thanks for the PR, let me find a reviewer for it

**Comment by davvd** [comment]
@yegor256 please review, thanks

File: s3auth-relay/src/main/java/com/s3auth/relay/HttpFacade.java [hunk 1]
```diff
@@ -256,6 +259,9 @@ public void run() {
             } catch (final InterruptedException ex) {
                 Thread.currentThread().interrupt();
                 Logger.warn(this, "%s", ex);
+            // @checkstyle IllegalCatch (1 line)
+            } catch (final Throwable ex) {
+                Logger.error(this, "%s", ExceptionUtils.getStackTrace(ex));
```
Comment (yegor256) [inline s3auth-relay/src/main/java/com/s3auth/relay/HttpFacade.java]: swallowing the exception? is it intentionally or by mistake?

File: s3auth-relay/src/main/java/com/s3auth/relay/HttpThread.java [hunk 2]
```diff
@@ -193,23 +195,24 @@ public long dispatch() throws InterruptedException {
...
                         HttpURLConnection.HTTP_BAD_METHOD,
                         "only GET and HEAD methods are supported at the moment"
                     ),
                     socket
                 );
-                Logger.info(this, "#run(): failure sent to %s", socket);
             }
         } catch (final HttpException ex) {
-            bytes = HttpThread.failure(ex, socket);
+            Logger.warn(this, "#run(): %s", ExceptionUtils.getStackTrace(ex));
```
Comment (yegor256) [inline s3auth-relay/src/main/java/com/s3auth/relay/HttpThread.java]: you can use a decorator of `Logger`, see http://log.jcabi.com/decors.html. in this case: `Logger.warn(this, "#run(): %[exception]s")`

File: s3auth-relay/src/main/java/com/s3auth/relay/HttpFacade.java [hunk 3]
```diff
@@ -256,6 +259,9 @@ public void run() {
             } catch (final InterruptedException ex) {
```
Comment (yegor256) [inline s3auth-relay/src/main/java/com/s3auth/relay/HttpFacade.java]: maybe we can use `VerboseRunnable` instead, see http://log.jcabi.com/threads-VerboseRunnable.html

**Comment by yegor256** [comment]
@alevohin see a few comments above

**Comment by alevohin** [comment]
@yegor256 please review one more time.

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@yegor256 Much obliged! I've added **18 mins** to your account in payment "000-bd942b39"... added +18 to your rating, now it is equal to [+4539](http://www.netbout.com/b/35008?open=rating)

**Comment by davvd** [comment]
@rultor deploy pls

**Pull request merged**